File: 3sem/4lab/Banks/BankAccounts/Factories/CreditBankAccountFactory.cs
using Banks.Clocks;
using Banks.Entities;

namespace Banks.BankAccounts.Factories;

public class CreditBankAccountFactory : IBankAccountFactory
{
    public IBankAccount Create(Bank bank, Client client, IClock clock)
    {
        return new CreditBankAccount(bank, client, clock);
    }
}

File: 4sem/1lab/src/main/java/BanksRemake/Domain/BankAccounts/Factories/CreditBankAccountFactory.java
package BanksRemake.Domain.BankAccounts.Factories;

import BanksRemake.Domain.BankAccounts.BankAccount;
import BanksRemake.Domain.BankAccounts.CreditBankAccount;
import BanksRemake.Domain.Clocks.Clock;
import BanksRemake.Domain.Entities.Bank;
import BanksRemake.Domain.Entities.Client;

/**
 * Factory creating credit bank accounts.
 */
public class CreditBankAccountFactory implements BankAccountFactory {
    @Override
    public BankAccount create(Bank bank, Client client, Clock clock) {
        return new CreditBankAccount(bank, client, clock);
    }
}
